Rust/src/samplesummary.rs

```diff
@@ -0,0 +1,374 @@
+use std::cmp::min;
+use num::abs;
+use rand::{Rng, SeedableRng};
+use rand_chacha::ChaCha20Rng;
+
+///
+/// The goal of the summarization below is as follows: on being provided a collection of sampled weighted points
+/// represented by a slice &[(Vec<f32>,f32)] where each of the Vec<f32> has the same length/dimension
+/// and the f32 in the pair is the corresponding weight.
+/// The algorithm uses the philosophy of RCFs, in repeatedly using randomization. It proceeds as follows:
+/// 1. It uses an initial sampling which serves as a basis of efficiency as well as denoising, borrowing from
+/// https://en.wikipedia.org/wiki/CURE_algorithm, in that algorithm's robustness to outliers.
+/// 2. It uses a sampling mechanism to initialize some clusters based on https://en.wikipedia.org/wiki/Data_stream_clustering
+/// where the radom sampling achieves half of the the same effects as hierarchical compression.
+///3.  It repeatedly merges the most overlapping clusters, failing that, eliminates the least weighted cluster to achieve
+/// the same effect as hieararchical compression.
+///
+/// The algorithm takes a distance function as an input, and tends to produce spherical (measured in the input
+/// distance function) clusters. These types of algorithms are unlikely to be useful for large number of output clusters.
+/// The output is the SampleSummary, which provides basic statistics of mean, median and deviation
+/// in addition it performs a grouping/clustering, assuming that the maximum number of clusters are not large
+/// the routine below bounds the number to be max_number_per_dimension times the dimension of Vec<f32>
+/// and a smaller number can also be provided in the summarize() function
+///
+///
+
+
+const max_number_per_dimension : usize = 5;
+
+#[repr(C)]
+pub struct SampleSummary {
+    pub summary_points: Vec<Vec<f32>>,
+
+    /**
+     * a measure of comparison among the typical points;
+     */
+    pub relative_weight: Vec<f32>,
+
+    /**
+     * number of samples, often the number of summary
```

Comment:
Incomplete sentence. If this is the number of samples, why is it a floating point value?

Reply:
Someday in the future -- we can pass points with confidence scores :) But that makes the case for the ConditionalSampleSummary to be a struct.